Content.Server/Light/EntitySystems/PoweredLightSystem.cs

```diff
@@ -103,6 +106,15 @@ private void OnInteractHand(EntityUid uid, PoweredLightComponent light, Interact
             if (bulbUid == null)
                 return;
 
+            // delay the interaction by 2 seconds
+            var curTime = _gameTiming.CurTime;
+            if (curTime < cooldownEnd)
+            {
+                return;
+            }
+
+            cooldownEnd = curTime + TimeSpan.FromSeconds(2);
+
```

Comment:
Well, this is definitely a cooldown.

It's a single cooldown for lights everywhere, which means that one person trying to burn their hand on a light at one end of the station can activate the very same cooldown that would be checked if someone completely different was trying to remove a lightbulb from a different light.

The reason why it works like this is because the `cooldownEnd` you added in the System is itself a part of the whole system. Normally, entity-specific things like this are added in variables on a Component, not a System. The components have data individual and specific to the entity they are attached to.

It's a good start though. If this is how we're going to resolve this, I'd suggest moving the cooldown check into the block below where it checks if it should burn the user, right before the `if (burnedHand)` line is probably good.

Reply:
oh i didnt knew it would make it affect affect to absolutely everyone, then i should move the cooldownEnd variable to the PoweredLightComponent, and i will move the check then.

Reply:
added the variable to the component and making as resolved :D